org.eclipse.lsp4xml/src/main/java/org/eclipse/lsp4xml/dom/XMLParser.java

```diff
@@ -189,12 +189,13 @@ public XMLDocument parse(TextDocument document, EnumSet<Flag> mask) {
 			case PIName: {
 				curr.tag = scanner.getTokenText();
 				curr.content = "";
+				((ProcessingInstruction) curr).setProcessingInstruction();
 				break;
 			}
 
 			case PrologName: {
 				curr.tag = scanner.getTokenText();
-				((ProcessingInstruction) curr).prolog = true;
+				((ProcessingInstruction) curr).setProlog();

```

Comment:
Please keep my original code if you ar eOK with my constants PROCESSING_INSTRUCTION_NODE. More I don't want to have setProlog and setProcessingInstruction to avoid external code update it. The only compoent which must manage those flag is the parser.